release-notes.txt

```diff
@@ -1,5 +1,540 @@
+RIOT-2024.07 "StrikinglyCrowded" - Release Notes
+================================================
+RIOT is a multi-threading operating system which enables soft real-time
+capabilities and comes with support for a range of devices that are typically
+found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
+light-weight 32-bit processors.
+
+RIOT is based on the following design principles: energy-efficiency, soft
+real-time capabilities, small memory footprint, modularity, and uniform API
+access, independent of the underlying hardware (with partial POSIX compliance).
+
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release
+==================
+
+The 2024.07 release, nicknamed "StrikinglyCrowded", includes many features and bug fixes.
+Here are some of the highlights:
+
++ RIOT now boots on the Arduino Nano 33 BLE Sense (#20668)
++ To support boards with Ethernet plugs such as the Adafruit Ethernet FeatherWing, RIOT now ships a driver for the W5500 ethernet chip (#20301)
++ RIOT now leverages the ARMv8-M hardware support to enforce the stack size (#20633)
++ The PSA Crypto API implementation in RIOT now supports persistent key storage and additional hashing algorithms (#20099, #20598, #20698)
++ This release features several useful additions to the RIOT CoAP stacks: a coaps forward proxy for gcoap, delayed responses for nanocoap_sock, and the ability to mount a remote filesystem over CoAP (#20454, #20266, #20687)
+* Support of the 16-bit MSP430 MCU has been improving over the last few releases, with the addition of power management in this release as an example (#20613)
+* First steps towards enabling RIOT support of new Microchip SAM MCUs (#20708, #20710, #20713, #20714)
+* Users of BUILD_IN_DOCKER=1 will now be warned if they are using an outdated docker container (#20472)
```

Comment:
```suggestion
* Compiling with `BUILD_IN_DOCKER=1` will now by default use the exact container version the code was tested in the CI, fetching it from upstream if not locally present. (#20472)
```

I forgot to update the PR description. The commit description does indeed match the behavior.